src/compactor_state.rs

```diff
@@ -138,11 +137,81 @@ impl CompactorState {
                 return Err(SlateDBError::InvalidCompaction);
             }
         }
+
+        self.validate_compaction(&compaction.sources, compaction.destination)?;
+
         info!("accepted submitted compaction: {:?}", compaction);
         self.compactions.insert(compaction.destination, compaction);
         Ok(())
     }
 
+    pub(crate) fn validate_compaction(
+        &mut self,
+        sources: &[SourceId],
+        destination: u32,
+    ) -> Result<(), SlateDBError> {
+        let compacted = &self.db_state.compacted;
+        let sources_logical_order: Vec<SourceId> = self
+            .db_state
+            .l0
+            .iter()
+            .map(|sst| SourceId::Sst(sst.id.unwrap_compacted_id()))
+            .chain(compacted.iter().map(|sr| SourceId::SortedRun(sr.id)))
+            .collect();
+
+        let sources_set: HashSet<_> = sources.iter().collect();
+
```

Comment:
Once you have `sources_logical_order`, a while loop is more readable and would be simpler.
something like
```rust
let mut index = 0;
while index < sources_logical_order.len() && sources_logical_order[index] != sources[0]{
index++;
}

// now we have the position of first source.
// iterate `sources` and `sources_logical_order` at the same time, they must match element by element.
// check destination using the two rules Rohan shared




```

Reply:
@vigneshc Sorry for the delay, I have been busy at work, I just did the changes you suggested, please let me know your feedback.